lib/ops/opsservice/site.go

```diff
@@ -644,6 +647,57 @@ func (s site) gid() string {
 	return defaults.ServiceUserID
 }
 
+func (s *site) getClusterConfiguration() (*clusterconfig.Resource, error) {
+	s.Info("Query existing cluster configuration.")
+	client, err := s.service.GetKubeClient()
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	configmap, err := client.CoreV1().ConfigMaps(defaults.KubeSystemNamespace).
+		Get(constants.ClusterConfigurationMap, metav1.GetOptions{})
+	err = rigging.ConvertError(err)
+	if err != nil && !trace.IsNotFound(err) {
+		return nil, trace.Wrap(err)
+	}
+	var spec string
+	if configmap != nil {
+		spec = configmap.Data["spec"]
+	}
+	var config *clusterconfig.Resource
+	if spec != "" {
+		config, err = clusterconfig.Unmarshal([]byte(spec))
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+		s.Infof("Existing cluster configuration: %#v (%s).", config, spec)
+	} else {
+		config = clusterconfig.NewEmpty()
+		s.Infof("Existing (empty) cluster configuration: %#v (%s).", config, spec)
+	}
+	if err := s.setClusterConfigDefaults(config); err != nil {
```

Comment:
This is to workaround issues with previous state after the upgrade.
Ideally, the installer would keep all now disparate attributes in the `ClusterConfiguration` from the get-go - which is something to improve in future (and I remember starting this work already for a related issue).